web_programming/emails_from_url.py

```diff
@@ -0,0 +1,97 @@
+"""Get the site emails from URL."""
+__author__ = "Muhammad Umer Farooq"
+__license__ = "MIT"
+__version__ = "1.0.0"
+__maintainer__ = "Muhammad Umer Farooq"
+__email__ = "contact@muhammadumerfarooq.me"
+__status__ = "Production"
+
+from html.parser import HTMLParser
+import requests
+import re
+from urllib.parse import urlparse
+
+from urllib import parse
+
+
+class Parser(HTMLParser):
+
+    def __init__(self, domain):
+        HTMLParser.__init__(self)
+        self.data = []
+        self.domain = domain
+
+    def handle_starttag(self, tag, attrs):
+        # Only parse the 'anchor' tag.
+        if tag == "a":
+            # Check the list of defined attributes.
+            for name, value in attrs:
+                # If href is defined, and not empty nor # print it.
+                if name == "href" and value != "#" and value != '':
+                    # If not already in data.
+                    if value not in self.data:
+                        url = parse.urljoin(self.domain, value)
+                        self.data.append(url)
+
+
+# Get main domain name (example.com)
+def get_domain_name(url):
+    try:
+        u = get_sub_domain_name(url).split('.')
+        return u[-2] + '.' + u[-1]
+    except:
```

Comment:
flake8 E722 do not use bare 'except' -- https://realpython.com/the-most-diabolical-python-antipattern

Reply:
https://travis-ci.com/TheAlgorithms/Python/builds/149132397#L360